rust/parquet_derive/src/parquet_field.rs

```diff
@@ -0,0 +1,790 @@
+            | Type::Reference(_, ref first_type) => match **first_type {
+                Type::TypePath(_) => &self,
+                Type::Option(ref second_type)
+                | Type::Vec(ref second_type)
+                | Type::Array(ref second_type)
+                | Type::Reference(_, ref second_type) => match **second_type {
+                    Type::TypePath(_) => first_type,
+                    Type::Option(ref third_type)
+                    | Type::Vec(ref third_type)
+                    | Type::Array(ref third_type)
+                    | Type::Reference(_, ref third_type) => match **third_type {
+                        Type::TypePath(_) => second_type,
+                        Type::Option(ref fourth_type)
+                        | Type::Vec(ref fourth_type)
+                        | Type::Array(ref fourth_type)
+                        | Type::Reference(_, ref fourth_type) => match **fourth_type {
+                            Type::TypePath(_) => third_type,
+                            _ => unimplemented!("sorry, I don't descend this far"),
+                        },
+                    },
+                },
+            },
+        }
+    }
+
+    pub fn last_part(&self) -> String {
+        let inner_type = self.inner_type();
+        let inner_type_str = (quote! { #inner_type }).to_string();
+
+        inner_type_str.split("::").last().unwrap().trim().to_string()
+    }
+
+    pub fn physical_type(&self) -> parquet::basic::Type {
+        use parquet::basic::Type as BasicType;
+
+        let last_part = self.last_part();
+        let leaf_type = self.leaf_type();
+
+        match leaf_type {
+            Type::Array(ref first_type) => {
+                if let Type::TypePath(_) = **first_type {
+                    if last_part == "u8" {
+                        return BasicType::FIXED_LEN_BYTE_ARRAY;
+                    }
+                }
+            }
+            Type::Vec(ref first_type) => {
+                if let Type::TypePath(_) = **first_type {
+                    if last_part == "u8" {
+                        return BasicType::BYTE_ARRAY;
+                    }
+                }
+            }
+            _ => (),
+        }
+
+        match last_part.trim() {
+            "bool" => BasicType::BOOLEAN,
+            "u8" | "u16" | "u32" => BasicType::INT32,
+            "i8" | "i16" | "i32" => BasicType::INT32,
+            "u64" | "i64" | "usize" | "NaiveDateTime" => BasicType::INT64,
+            "f32" => BasicType::FLOAT,
+            "f64" => BasicType::DOUBLE,
+            "String" | "str" => BasicType::BYTE_ARRAY,
+            f @ _ => unimplemented!("sorry, don't handle {} yet!", f),
+        }
+    }
+
+    pub fn from_type_path(f: &syn::Field, p: &syn::TypePath) -> Self {
+        let segments_iter = p.path.segments.iter();
+        let segments: Vec<syn::PathSegment> = segments_iter.map(|x| x.clone()).collect();
+        let last_segment = segments.last().unwrap();
```

Comment:
Can this be simplified to the following?
```rust
        let last_segment = p.path.segments.last().unwrap().into_value();
```

Reply:
Thank you, good catch!